client/src/components/Availability/Availability.tsx

```diff
@@ -0,0 +1,139 @@
+import { ComponentType, useState, useEffect } from 'react';
+import { styled } from '@mui/material/styles';
+import TextField from '@mui/material/TextField';
+import AdapterDateFns from '@mui/lab/AdapterDateFns';
+import LocalizationProvider from '@mui/lab/LocalizationProvider';
+import PickersDay, { PickersDayProps } from '@mui/lab/PickersDay';
+import DesktopDatePicker from '@mui/lab/DesktopDatePicker';
+import endOfWeek from 'date-fns/endOfWeek';
+import isSameDay from 'date-fns/isSameDay';
+import isWithinInterval from 'date-fns/isWithinInterval';
+import startOfWeek from 'date-fns/startOfWeek';
+import format from 'date-fns/format';
+import { addDays, addMonths, differenceInDays, differenceInMonths } from 'date-fns';
+import Typography from '@material-ui/core/Typography';
+import Box from '@material-ui/core/Box';
+import useStyles from './useStyles';
+import Grid from '@material-ui/core/Grid';
+import Card from '@material-ui/core/Card';
+import TimePicker from './TimePicker/TimePicker';
+
+type CustomPickerDayProps = PickersDayProps<Date> & {
+  dayIsBetween: boolean;
+  isFirstDay: boolean;
+  isLastDay: boolean;
+};
+
+const CustomPickersDay = styled(PickersDay, {
+  shouldForwardProp: (prop) => prop !== 'dayIsBetween' && prop !== 'isFirstDay' && prop !== 'isLastDay',
+})<CustomPickerDayProps>(({ theme, dayIsBetween, isFirstDay, isLastDay }) => ({
+  ...(dayIsBetween && {
+    borderRadius: 0,
+    backgroundColor: theme.palette.primary.main,
+    color: theme.palette.common.white,
+    '&:hover, &:focus': {
+      backgroundColor: theme.palette.primary.dark,
+    },
+  }),
+  ...(isFirstDay && {
+    borderTopLeftRadius: '50%',
+    borderBottomLeftRadius: '50%',
+  }),
+  ...(isLastDay && {
+    borderTopRightRadius: '50%',
+    borderBottomRightRadius: '50%',
+  }),
+})) as ComponentType<CustomPickerDayProps>;
+
+function dateRange(startDate: Date, endDate: Date) {
+  const days = differenceInDays(endDate, startDate);
+
+  return [...Array(days + 1).keys()].map((i) => addDays(startDate, i));
+}
+
+export default function CustomDay() {
+  const [value, setValue] = useState<Date | null>(new Date());
+  const [timeValue, setTimeValue] = useState<Date | null>(new Date('2018-01-01T00:00:00.000Z'));
+
+  const [dates, setDates] = useState<Array<Date | Null>>([]);
```

Comment:
Having the availability dates here from a sitter's profile would be ideal (currently, if the page is refreshed, the sitter will not be able to determine the time slots they've set) 😄